instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/cache/Cache.java

```diff
@@ -3,17 +3,34 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-package io.opentelemetry.instrumentation.api.caching;
+package io.opentelemetry.instrumentation.api.cache;
 
+import io.opentelemetry.instrumentation.api.cache.concurrentlinkedhashmap.ConcurrentLinkedHashMap;
```

Comment:
Should we put CLHM and WeakLockFree in `internal` package(s)? They shouldn't be a part of our public API.

Reply:
👍 created #4735 to track